### DISCUSSION TITLE
Help needed with yabai/skhd config

### DISCUSSION THREAD

**Discussion by justsharan** [asmvik/yabai#740] [Q&A] [answered]
Hello! I'd like to define a hotkey that moves west if the current window is to the east and east if the current window is to the west. So it basically toggles between east/west. How would I do that?

**Comment by dominiklohmann** [comment, marked as answer]
You can implement wraparound like this:

```
# cycle forwards
yabai -m window --focus next || yabai -m window --focus first

# cycle backwards
yabai -m window --focus prev || yabai -m window --focus last
```

Or, for a more complex solution that works a fair bit better:

```
# cycle forwards
yabai -m query --spaces --space \
  | jq -re ".index" \
  | xargs -I{} yabai -m query --windows --space {} \
  | jq -sre "add | map(select(.minimized != 1)) | sort_by(.display, .frame.y, .frame.x, .id) | reverse | nth(index(map(select(.focused == 1))) - 1).id" \
  | xargs -I{} yabai -m window --focus {}

# cycle backwards
yabai -m query --spaces --space \
  | jq -re ".index" \
  | xargs -I{} yabai -m query --windows --space {} \
  | jq -sre "add | map(select(.minimized != 1)) | sort_by(.display, .frame.y, .frame.x, .id) | nth(index(map(select(.focused == 1))) - 1).id" \
  | xargs -I{} yabai -m window --focus {}
```

**Reply by rockyzhang24** [reply]
In the easier one, `|` should be `||`, right?

**Reply by dominiklohmann** [reply]
You are absolutely correct. I edited the post accordingly.

**Reply by rockyzhang24** [reply]
For the complex solution, can I simplify it to:
```bash
# cycle forwards
yabai -m query --windows --space \
  | jq -re "map(select(.minimized != 1)) | sort_by(.frame.y, .frame.x, .id) | reverse | nth(index(map(select(.focused == 1))) - 1).id" \
  | xargs -I{} yabai -m window --focus {}

# cycle backwards
yabai -m query --windows --space \
  | jq -re "map(select(.minimized != 1)) | sort_by(.frame.y, .frame.x, .id) | nth(index(map(select(.focused == 1))) - 1).id" \
  | xargs -I{} yabai -m window --focus {}
```

(1). We only care about all the windows in the currently focused workspace, so we can get all the windows in the space by `yabai -m query --windows --space` directly.
(2). I removed `-s` along with `add`. I think we can directly use `map()` first. 
(3). I removed `.display` in `sort_by()`, because we're caring about windows in the focused workspace, thus nothing to do with other displays.

Am I correct? Thanks a lot.

**Reply by ajc5660** [reply]
Amazing solution
